build/source/engine/allocspace.f90

```diff
@@ -244,6 +258,258 @@ subroutine allocLocal(metaStruct,dataStruct,nSnow,nSoil,err,message)
  end subroutine allocLocal
 
  ! ************************************************************************************************
+ ! public subroutine resizeData: resize data structure
+ ! ************************************************************************************************
+ subroutine resizeData(metaStruct,dataStructOrig,dataStructNew,copy,err,message)
```

Comment:
The `allocate(data, mold=)` assumes arrays start at 1. Hence we needed to write a home-grown re-sizer. Perhaps in the future the compilers will support what we desire, but for now we need to keep the code as it is.